worlds/marioland2/__init__.py

```diff
@@ -0,0 +1,372 @@
+                   },
+        "Normal Exits": {location for location in locations if locations[location]["type"] == "level"},
+        "Secret Exits": {location for location in locations if locations[location]["type"] == "secret"},
+        "Bells": {location for location in locations if locations[location]["type"] == "bell"},
+    }
+
+    option_definitions = sml2options
+
+    generate_output = generate_output
+
+    def create_regions(self):
+        menu_region = Region("Menu", self.player, self.multiworld)
+        self.multiworld.regions.append(menu_region)
+        created_regions = []
+        for location_name, data in locations.items():
+            if "Secret Course" in location_name:
+                region_name = location_name
+            elif "Mushroom Zone" in location_name:
+                region_name = "Mushroom Zone"
+            else:
+                region_name = location_name.split(" -")[0]
+            if region_name in created_regions:
+                region = self.multiworld.get_region(region_name, self.player)
+            else:
+                region = Region(region_name, self.player, self.multiworld)
+                if location_name == "Tree Zone - Secret Course":
+                    region_to_connect = self.multiworld.get_region("Tree Zone 2", self.player)
+                elif location_name == "Space Zone - Secret Course":
+                    region_to_connect = self.multiworld.get_region("Space Zone 1", self.player)
+                elif location_name == "Macro Zone - Secret Course":
+                    region_to_connect = self.multiworld.get_region("Macro Zone 1", self.player)
+                elif location_name == "Pumpkin Zone - Secret Course 1":
+                    region_to_connect = self.multiworld.get_region("Pumpkin Zone 2", self.player)
+                elif location_name == "Pumpkin Zone - Secret Course 2":
+                    region_to_connect = self.multiworld.get_region("Pumpkin Zone 3", self.player)
+                elif location_name == "Turtle Zone - Secret Course":
+                    region_to_connect = self.multiworld.get_region("Turtle Zone 2", self.player)
+                elif "-" in location_name and int(location_name.split(" ")[2]) > 1:
+                    region_to_connect = self.multiworld.get_region(" ".join(location_name.split(" ")[:2])
+                                                                   + f" {int(location_name.split(' ')[2]) - 1}",
+                                                                   self.player)
+                else:
+                    region_to_connect = menu_region
+                region_to_connect.connect(region)
+                self.multiworld.regions.append(region)
+                created_regions.append(region_name)
+
+            if "Midway Bell" in location_name and not self.multiworld.shuffle_midway_bells[self.player]:
```

Comment:
```suggestion
            if "Midway Bell" in location_name and not self.shuffle_midway_bells:
```
Should swap to using the new options API